src/stub/core_impl/base.rs

```diff
@@ -202,7 +202,7 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
                 regs.gdb_serialize(|val| {
                     let res = match val {
                         Some(b) => res.write_hex_buf(&[b]),
-                        None => res.write_str("xx"),
+                        None => res.write_x(std::mem::size_of::<u8>()),
```

Comment:
Ahh, due to the way I structured CI, the no_std conformance check will only run after clippy succeeds.

In any case, we can't use `std::` in a no_std project, so you'll need to switch this out to use `core`.

(though as mentioned in a previous comment - I'm still wary of introducing this write_x helper, given that it only gets used twice)

Reply:
Re `std::`: Good catch!